conversion.go

```diff
@@ -133,6 +133,10 @@ func ToInt64[T Number](i T) (int64, error) {
 		return 0, fmt.Errorf("%w: %v is greater than math.MaxInt64", ErrOutOfRange, i)
 	}
 
+	if float64(i) >= float64(math.MaxInt64) {
+		return 0, fmt.Errorf("%w: %v is greater than math.MaxInt64", ErrOutOfRange, i)
+	}
+
```

Comment:
wip? missing negative max range check
(and the other types, I think you'd want to type check/type dispatch ints and floats differently... not sure)

Reply:
This one is a pain because a float64 can be way bigger than MaxInt64

converting it  to a float help

but yes, it was a left behind wip

I'm closing, please review again the new code

Please note, you may open the same issues, but the discussion won't continue on outdated code